agent/acs/handler/refresh_credentials_handler.go

```diff
@@ -125,18 +125,25 @@ func (refreshHandler *refreshCredentialsHandler) handleSingleMessage(message *ec
 	task, ok := refreshHandler.taskEngine.GetTaskByArn(taskArn)
 	if !ok {
 		seelog.Errorf("Task not found in the engine for the arn in credentials message, arn: %s, messageId: %s", taskArn, messageId)
-		return fmt.Errorf("Task not found in the engine for the arn in credentials message, arn: %s", taskArn)
+		return fmt.Errorf("task not found in the engine for the arn in credentials message, arn: %s", taskArn)
 	}
-	taskCredentials := credentials.TaskIAMRoleCredentials{
-		ARN:                taskArn,
-		IAMRoleCredentials: credentials.IAMRoleCredentialsFromACS(message.RoleCredentials),
-	}
-	err = refreshHandler.credentialsManager.SetTaskCredentials(taskCredentials)
-	if err != nil {
-		seelog.Errorf("Error updating credentials, err: %v messageId: %s", err, messageId)
-		return fmt.Errorf("Error updating credentials %v", err)
+
+	roleType := aws.StringValue(message.RoleType)
+	if !validRoleType(roleType) {
+		seelog.Errorf("Unknown RoleType for task in credentials message, roleType: %s arn: %s, messageId: %s", roleType, taskArn, messageId)
+	} else {
+
+		taskCredentials := credentials.TaskIAMRoleCredentials{
+			ARN:                taskArn,
+			IAMRoleCredentials: credentials.IAMRoleCredentialsFromACS(message.RoleCredentials, roleType),
+		}
+		err = refreshHandler.credentialsManager.SetTaskCredentials(taskCredentials)
+		if err != nil {
+			seelog.Errorf("Unable to update credentials for task, err: %v messageId: %s", err, messageId)
+			return fmt.Errorf("unable to update credentials %v", err)
+		}
+		task.SetCredentialsID(aws.StringValue(message.RoleCredentials.CredentialsId))
```

Comment:
`SetCredentialsID` is used to set the id for task iam role, where we have two type of roles now. You may need to set this based on the type of the role.

Reply:
you're right, ill add this check.